### PR TITLE
refactor(DataList): Rename to `DescriptionList`

### LINKED ISSUE #2423: Rename `Data List` to `Description List` to better match HTML spec
We currently have a `Data List` component that does not match the [Data List spec](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist). Instead, the component we have uses `<dl>`, `<dt>` and `<dd>` tags, which are specific to [Description Lists](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl).

- Recommendation is to change the name of the `Data List` component to `Description List`, to be better match the functionality and behaviour of said component.

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #2423

## Overview

Rename component to better match W3C spec.

### CODE REVIEW

**Review by thyhjwb6 — APPROVED** [review]
Looks good 👍 

I have been taking the opportunity to add a Cypress test in too....

File: packages/react-component-library/src/components/DescriptionList/DescriptionList.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import { ComponentStory, ComponentMeta } from '@storybook/react'
+
+import { DescriptionList, DescriptionListItem } from './index'
+
+export default {
+  component: DescriptionList,
+  subcomponents: { DescriptionListItem },
+  title: 'Data List',
```
Comment (jpveooys) [inline packages/react-component-library/src/components/DescriptionList/DescriptionList.stories.tsx]: Think this needs updating:

```suggestion
  title: 'Desscription List',
```

File: packages/react-component-library/src/components/DescriptionList/DescriptionList.test.tsx [hunk 2]
```diff
@@ -2,21 +2,21 @@ import React from 'react'
...
-          </DataListItem>
-          <DataListItem title="Two">2</DataListItem>
-          <DataListItem title="Three">3</DataListItem>
-        </DataList>
+          </DescriptionListItem>
+          <DescriptionListItem title="Two">2</DescriptionListItem>
+          <DescriptionListItem title="Three">3</DescriptionListItem>
+        </DescriptionList>
       )
     })
```
Comment (jpveooys) [inline packages/react-component-library/src/components/DescriptionList/DescriptionList.test.tsx:22]: Do the `data list`s in the `it()`s below need updating?

File: packages/react-component-library/src/components/DescriptionList/DescriptionList.tsx [hunk 3]
```diff
@@ -54,7 +54,7 @@ export const DataList: React.FC<DataListProps> = ({
   const sheetId = ariaAttributes && ariaAttributes['aria-owns']
 
   return (
-    <StyledDataList
+    <StyledDescriptionList
       $isCollapsible={isCollapsible}
       $isOpen={open}
       data-testid="data-list"
```
Comment (jpveooys) [inline packages/react-component-library/src/components/DescriptionList/DescriptionList.tsx]: Do the test IDs need updating? (This one and the two below.)

File: packages/react-component-library/src/components/DescriptionList/DescriptionListItem.tsx [hunk 4]
```diff
@@ -20,7 +20,7 @@ export interface DataListItemProps {
   isCollapsible?: boolean
 }
 
-export const DataListItem: React.FC<DataListItemProps> = ({
+export const DescriptionListItem: React.FC<DescriptionListItemProps> = ({
   children,
   title,
   isCollapsible,
```
Comment (jpveooys) [inline packages/react-component-library/src/components/DescriptionList/DescriptionListItem.tsx:26]: Does the test ID below need updating?

**Review by jpveooys — COMMENTED** [review]
Just a few small bits that look like they need updating too.

**Review by jpveooys — APPROVED** [review]

**Pull request merged**